dotCMS/src/main/java/com/dotcms/ema/EMAWebInterceptor.java

```diff
@@ -0,0 +1,208 @@
+
+        final Host currentHost = getCurrentHost(request);
+
+        if (!this.existsConfiguration(currentHost.getIdentifier())) {
+            return Result.NEXT;
+        }
+
+        final Optional<String> proxyUrl = proxyUrl(currentHost);
+        final PageMode mode             = PageMode.get(request);
+
+        if (!proxyUrl.isPresent() || mode == PageMode.LIVE) {
+            return Result.NEXT;
+        }
+
+        Logger.info(this.getClass(), "GOT AN EMA Call --> " + request.getRequestURI());
+
+        return new Result.Builder().wrap(new MockHttpCaptureResponse(response)).next().build();
+    }
+
+
+    @Override
+    public boolean afterIntercept(final HttpServletRequest request, final HttpServletResponse response) {
+
+        try {
+
+            if (response instanceof MockHttpCaptureResponse) {
+
+                final Host currentHost = getCurrentHost(request);
+                final Optional<String> proxyUrl            = proxyUrl(currentHost);
+                final MockHttpCaptureResponse mockResponse = (MockHttpCaptureResponse)response;
+                final String postJson                      = new String(mockResponse.getBytes());
+                final JSONObject json                      = new JSONObject(postJson);
+                final Map<String, String> params           = ImmutableMap.of("dotPageData", postJson);
+                
+                Logger.info(this.getClass(), "Proxying Request --> " + proxyUrl.get());
+                
+                String responseStr = new String();
+                final ProxyResponse pResponse = proxy.sendPost(proxyUrl.get(), params);
+
+                if (pResponse.getResponseCode() == 200) {
+                    responseStr = new String(pResponse.getResponse());
+                }else {
+                    responseStr+="<html><body>";
+                    responseStr+="<h3>Unable to connect with the rendering engine</h3>";
+                    responseStr+="<br><div style='display:inline-block;width:80px'>Trying: </div><b>" + proxyUrl.get()  + "</b>";
+                    responseStr+="<br><div style='display:inline-block;width:80px'>Got:</div><b>" + pResponse.getStatus() + "</b>";
+                    responseStr+="<hr>";
+                    responseStr+="<h4>Headers</h4>";
+                    responseStr+="<table border=1 style='min-width:500px'>";
+
+                    for(Header header : pResponse.getHeaders()) {
+                      responseStr+="<tr><td style='font-weight:bold;padding:5px;'><pre>" + header.getName() + "</pre></td><td><pre>" + header.getValue() + "</td></tr>";
+                    }
+                    responseStr+="</table>";
+                    
+                    responseStr+="<p>The Json Payload, POSTing as Content-Type:'application/x-www-form-urlencoded' with form param <b>dotPageData</b>, has been printed in the logs.</p>";
+                    responseStr+="</body></html>";
```

Comment:
Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)